pkg/v3/types/basetypes.go

```diff
@@ -216,12 +234,52 @@ func (r CheckResult) UniqueID() string {
 
 func (r CheckResult) String() string {
 	return fmt.Sprintf(
-		checkResultStringTemplate, r.PipelineExecutionState, r.Retryable, r.RetryInterval, r.Eligible,
```

Comment:
why remove from here?

Reply:
To avoid ambiguity, we use `String()` to encode into JSON, I don't want it to get into a report somehow

Reply:
hmm i see. still confusing for me personally but i guess any solution will have some level of confusion.

Maybe add a comment on top of this function as to how this should be modified? (e.g. we want to keep it consistent with marshallJSON?)

Reply:
Do we have a good reason why not to reduce that String function? i.e. to keep the standard golang printing of that object? If its just the hex encoding of PerformData we can use a custom type for it and reduce the function completely

Reply:
yup agree this can be potentially cleaned up